src/chunk.c

```diff
@@ -837,6 +850,54 @@ chunk_find(Hyperspace *hs, Point *p)
 	return chunk;
 }
 
+List *
+chunk_find_all_oids(Hyperspace *hs, List *dimension_vecs, LOCKMODE lockmode)
+{
+	List	   *chunk_list,
+			   *oid_list;
+	ChunkScanCtx ctx;
+	ListCell   *lc;
+
+	/* The scan context will keep the state accumulated during the scan */
+	chunk_scan_ctx_init(&ctx, hs, NULL);
+
+	/* Abort the scan when the chunk is found */
+	ctx.early_abort = false;
```

Comment:
Comment & code don't match, at least I don't think. It appears that you are _not_ ending the scan when the chunk is found.